### DISCUSSION TITLE
File size too big - ERR_HTTP2_PING_FAILED

### DISCUSSION THREAD

**Discussion by Irv007** [nginx-proxy/nginx-proxy#2154] [Q&A] [answered]
Hi,

I'm managing my e-books with Calibre-Web and Nginx. Since about two weeks, I'm not be able to upload books to Calibre-Web any longer. This has been working fine for several years.

Each uploaded book now results in: "Error: File size may be too big". Even if the book is just 500 K.

So, that has nothing to do with the nginx setting "client_max_body_size", the default here is "1M".

The message comes from Calibre Web, but Calibre Web just reacts to a problem, which occured previously.

The error occurs in Google, Edge and Firefox on Windows 10.
The error does not occur from Safari on iOS.

The message in Google Chrome Developer Console is:

====>
```
POST https://xbuch.eu/upload
uploadprogress.js:99           
net::ERR_HTTP2_PING_FAILED

submit          @ uploadprogress.js:99
i               @ jquery.min.js:2
dispatch        @ jquery.min.js:2
v.handle        @ jquery.min.js:2
trigger         @ jquery.min.js:2
(anonymous)     @ jquery.min.js:2
each            @ jquery.min.js:2
each            @ jquery.min.js:2
trigger         @ jquery.min.js:2
S.fn.<computed> @ jquery.min.js:2
(anonymous)     @ main.js:142
dispatch        @ jquery.min.js:2
v.handle        @ jquery.min.js:2
```
====>

The calibre-web people did check this and pointed me to my proxy, which is nginx (nginxproxy/nginx-proxy on docker) (https://github.com/janeczku/calibre-web/issues/2678)

**Environment**

 - Server:
 - OS: 20.04.1-Ubuntu on Google Cloud Platform
 - Calibre-Web version: 0.6.19
 - Docker container: linuxserver/calibre-web, nginxproxy/nginx-proxy

 - Client:
 - OS: Windows 10
 - Browser: Google Chrome, Microsoft Edge, Firefox

with best regards,
Irvin
<img width="1152" alt="docker-compose yml" src="https://user-images.githubusercontent.com/35102767/215267078-1a0f75e1-5b58-41d5-a53a-4c1d9c8560aa.png">

**Comment by cansurmeli** [comment]
Hey,

I saw your indication of file sizes within your custom configuration file with `client_max_body_size`. I had the same problem with Nextcloud on my setup. Although the file sizes were much bigger in my case within my setup I mounted the configuration file as below for the nginx_proxy container:

`./my_proxy.conf:/etc/nginx/conf.d/my_proxy.conf:ro`

Could you mount it like this, then set the limit to something bigger like 10M perhaps and then report back please.

**Comment by Irv007** [comment]
ok, did that.

stated separate nginx.conf in docker-compose.yml ("- ./nginx.conf:/etc/nginx/nginx.conf" in "volumes:")
nginx.conf http-block: client_max_body_size 10M;

Restarted containers.

Nothing changed.

**Comment by cansurmeli** [comment]
For the container volume, did you also wrote the `conf.d` folder between `nginx` and `nginx.conf` or did you simply forget while writing the response?

Also, perhaps I should have written this directly in my initial response but my configuration file is merely a line like below:

`client_max_body_size 10M;`

**Comment by Irv007** [comment]
I changed my configuration like below. Restarted containers (docker-compose down; docker-compose up -d;). No difference.

![yml extract](https://user-images.githubusercontent.com/35102767/215277558-e7f8b0d4-a723-4383-8246-dd368a153879.png)
![nginx conf](https://user-images.githubusercontent.com/35102767/215277562-68bdcb5c-ad4e-4a57-a277-e876daf7c903.png)

**Comment by Irv007** [comment]
from ios, it works like a charm

**Comment by Irv007** [comment, marked as answer]
- I changed nginx to traefik.
- Same problem.
- Haha....
- nothing todo with nginx.
- Just tried on my working laptop ... everything ist ok.
- So it's somewithing with my client windows pc.
- this was the reason, why it worked from ios
- it works from every client pc except my main home windows pc.
- It even works on my client windows pc via WLAN, but not via ethernet.
- So, it's something with my ethernet network card.

Topic closed.
